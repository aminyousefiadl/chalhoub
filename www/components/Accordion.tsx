"use client";

import React, { useState, useMemo, useCallback } from "react";
import "rc-slider/assets/index.css";
import Slider from "rc-slider";
import { useConfig } from "@/context/information.context";
import { ColorsList } from "@/constants/Colors";

enum FiltersTitle {
  Color = "Color",
  Price = "Price",
}

export const Accordion = ({
  defaultExpanded,
}: {
  defaultExpanded?: "price" | "color";
}) => {
  const [isColorExpanded, setColorExpanded] = useState(
    defaultExpanded === "color"
  );
  const [isPriceExpanded, setPriceExpanded] = useState(
    defaultExpanded === "price"
  );

  const {
    setInformation,
    information: { color, priceRange },
  } = useConfig();

  const handleExpandClick = useCallback((el: FiltersTitle) => {
    if (el === FiltersTitle.Color) setColorExpanded((prev) => !prev);
    else if (el === FiltersTitle.Price) setPriceExpanded((prev) => !prev);
  }, []);

  const colorContent = useMemo(
    () => (
      <div className="py-9">
        {color && (
          <span
            className="cursor-pointer mx-2 text-primary text-sm"
            onClick={() => setInformation({ color: "" })}
          >
            Clear
          </span>
        )}
        <div className="flex flex-wrap">
          {ColorsList.map((el) => (
            <span
              key={el}
              onClick={() => setInformation({ color: el })}
              className={`cursor-pointer w-6 flex m-2 h-6 shadow-lg rounded-sm ${
                color === el ? "border-2 border-blue-500" : ""
              }`}
              style={{ backgroundColor: el.toLocaleLowerCase() }}
            />
          ))}
        </div>
      </div>
    ),
    [color, setInformation]
  );

  const priceContent = useMemo(
    () => (
      <div className="p-3 border border-b-0 border-gray-200  py-9">
        <p className="text-sm mb-2 text-primary">
          <span className="mr-2 text-gray-400 text-xs">From</span>$
          {priceRange[0].toLocaleString()}
          <span className="mx-2 text-gray-400 text-xs">To</span>$
          {priceRange[1].toLocaleString()}
        </p>
        <Slider
          range
          min={0}
          max={20000}
          value={priceRange}
          onChange={(e) => setInformation({ priceRange: e as number[] })}
        />
      </div>
    ),
    [priceRange, setInformation]
  );

  return (
    <div id="accordion-collapse" data-accordion="collapse">
      {Object.values(FiltersTitle).map((el: FiltersTitle) => (
        <React.Fragment key={el}>
          <h2 id="accordion-collapse-heading-1">
            <button
              name={`expandable-div-${el}`}
              type="button"
              className="flex items-center justify-between w-full p-5 font-medium text-left text-gray-500  border-b border-gray-200    hover:bg-gray-100 "
              data-accordion-target="#accordion-collapse-body-1"
              aria-expanded={
                el === FiltersTitle.Color ? isColorExpanded : isPriceExpanded
              }
              aria-controls="accordion-collapse-body-1"
              onClick={() => handleExpandClick(el)}
            >
              <span>{el}</span>

              <svg
                data-accordion-icon
                className={`w-3 h-3 transition-[transform] duration-500 ease-in ${
                  el === FiltersTitle.Color
                    ? isColorExpanded
                      ? ""
                      : "rotate-180"
                    : isPriceExpanded
                    ? ""
                    : "rotate-180"
                } shrink-0`}
                aria-hidden="true"
                xmlns="http://www.w3.org/2000/svg"
                fill="none"
                viewBox="0 0 10 6"
              >
                <path
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  d="M9 5 5 1 1 5"
                />
              </svg>
            </button>
          </h2>
          <div
            id="accordion-collapse-body-1"
            className={`${
              el === FiltersTitle.Color
                ? isColorExpanded
                  ? "max-h-60"
                  : "max-h-0"
                : isPriceExpanded
                ? "max-h-60"
                : "max-h-0"
            } transition-[max-height] duration-500 ease-in  overflow-hidden `}
            aria-labelledby="accordion-collapse-heading-1"
          >
            {el === FiltersTitle.Color ? colorContent : priceContent}
          </div>
        </React.Fragment>
      ))}
    </div>
  );
};
