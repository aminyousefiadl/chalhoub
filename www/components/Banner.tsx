import Image from "next/image";

export const Banner = () => {
  return (
    <section className="relative w-full h-[480px] mb-8">
      <Image
        className="rounded-lg"
        src="/banner_promotion.webp"
        alt="Promotion Banner"
        layout="fill"
      />
    </section>
  );
};
