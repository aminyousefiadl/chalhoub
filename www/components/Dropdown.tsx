"use-client";
import React from "react";

interface DropdownProps {
  options: string[];
  value: string;
  onChange: React.ChangeEventHandler<HTMLSelectElement>;
}

export const Dropdown: React.FC<DropdownProps> = ({
  options,
  value,
  onChange,
}) => {
  return (
    <select
      id="select"
      className="bg-gray-100 min-w-[6rem] border max-w-xl border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full py-2.5 px-4  "
      onChange={onChange}
      value={value}
      placeholder="Select"
    >
      {options.map((el: string) => (
        <option key={el} value={el}>
          {el}
        </option>
      ))}
    </select>
  );
};
