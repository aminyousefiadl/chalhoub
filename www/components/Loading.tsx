export const Loading = () => {
  return (
    <div className="grid grid-cols-1 md:grid-cols-4 gap-4 md:gap-8">
      {Array(12)
        .fill("")
        .map((_, i) => (
          <div
            key={i}
            className="rounded-lg animate-pulse bg-gray-300 h-48 md:h-[356px]"
          />
        ))}
    </div>
  );
};
