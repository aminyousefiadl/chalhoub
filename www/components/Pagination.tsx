import { useConfig } from "@/context/information.context";

export const Pagination = () => {
  const {
    setInformation,
    information: { page, limit },
    products,
  } = useConfig();

  if (
    (limit !== "All" && products?.data.length !== 0, products?.totalPages === 1)
  )
    return null;

  return (
    <nav
      aria-label="Page navigation example "
      className="flex justify-center mt-8"
    >
      <ul className="inline-flex -space-x-px text-sm">
        <li
          className="cursor-pointer flex items-center justify-center px-3 h-8 ml-0 leading-tight text-gray-500 bg-white border border-gray-300 rounded-l-lg hover:bg-gray-100 hover:text-gray-700  "
          onClick={() => setInformation({ page: page !== 1 ? page - 1 : 1 })}
        >
          Previous
        </li>

        {Array(products?.totalPages)
          .fill("")
          .map((_el, i) => (
            <li
              key={i}
              onClick={() => setInformation({ page: i + 1 })}
              className={`${
                page === i + 1 ? "bg-gray-200 text-primary" : ""
              } cursor-pointer flex items-center justify-center px-3 h-8 leading-tight text-gray-500 bg-white border border-gray-300 hover:bg-gray-100 hover:text-gray-700  `}
            >
              {i + 1}
            </li>
          ))}
        <li
          onClick={() => setInformation({ page: page !== 5 ? page + 1 : 5 })}
          className="cursor-pointer flex items-center justify-center px-3 h-8 leading-tight text-gray-500 bg-white border border-gray-300 rounded-r-lg hover:bg-gray-100 hover:text-gray-700  "
        >
          Next
        </li>
      </ul>
    </nav>
  );
};
