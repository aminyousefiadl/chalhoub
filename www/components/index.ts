export * from "./Accordion";
export * from "./Tile";
export * from "./Dropdown";
export * from "./Sort";
export * from "./Pagination";
export * from "./Filter";
export * from "./Banner";