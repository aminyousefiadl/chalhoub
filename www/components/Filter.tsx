"use client";

import { Dispatch, SetStateAction, useState } from "react";
import { Accordion } from "./Accordion";

const CloseIcon = ({
  setOpen,
}: {
  setOpen: Dispatch<SetStateAction<boolean>>;
}) => (
  <svg
    xmlns="http://www.w3.org/2000/svg"
    fill="none"
    viewBox="0 0 17 14"
    className="w-6 h-6 text-gray-800  absolute left-4 top-4 cursor-pointer"
    onClick={() => setOpen(true)}
  >
    <path
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      d="M1 1h15M1 7h15M1 13h15"
    />
  </svg>
);

export const Filter = () => {
  const [open, setOpen] = useState(false);

  return (
    <>
      <CloseIcon setOpen={setOpen} />
      <div
        className={`md:transition-[transform] md:h-[100vh] duration-700 ease-in-out bg-white absolute md:sticky top-2 w-full h-full rounded-lg z-20 md:w-56 shadow-md ${
          open ? "translate-y-0" : "translate-y-full md:translate-y-0"
        }`}
      >
        <span
          onClick={() => setOpen(false)}
          className="cursor-pointer md:hidden m-4 text-white bg-black rounded-full w-6 h-6 flex items-center justify-center"
        >
          X
        </span>
        <h5 className="font-bold flex p-4">
          <svg
            xmlns="http://www.w3.org/2000/svg"
            fill="none"
            viewBox="0 0 24 24"
            strokeWidth={1.5}
            stroke="currentColor"
            className="w-6 h-6 mr-2"
          >
            <path
              strokeLinecap="round"
              strokeLinejoin="round"
              d="M3.75 6.75h16.5M3.75 12h16.5m-16.5 5.25H12"
            />
          </svg>
          Filters
        </h5>
        <hr />
        <Accordion defaultExpanded="color" />
      </div>
    </>
  );
};
