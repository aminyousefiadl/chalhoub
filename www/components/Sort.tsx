"use client";

import { useConfig } from "@/context/information.context";
import { Dropdown } from "./Dropdown";
import { DisplayColsOptions, SortOptions } from "@/types";

export const Sort = () => {
  const {
    setInformation,
    information: { displayCols, limit, sortType },
  } = useConfig();

  const handleDisplayColsChange = (cols: string) => {
    setInformation({ displayCols: cols as DisplayColsOptions });
  };

  const handleLimitChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
    setInformation({ limit: e.target.value });
  };

  const handleSortTypeChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
    setInformation({ sortType: e.target.value as SortOptions });
  };

  return (
    <div className="bg-white md:sticky top-0 rounded-b-lg z-20 mt-16 md:mt-0 md:flex items-center justify-between p-4 w-full shadow-lg mb-8">
      <div className="md:flex items-center">
        <div className="flex items-center mr-8">
          {["4", "6", "8"].map((cols) => (
            <svg
              key={cols}
              onClick={() => handleDisplayColsChange(cols)}
              xmlns="http://www.w3.org/2000/svg"
              fill="none"
              viewBox="0 0 24 24"
              strokeWidth={1.5}
              stroke="currentColor"
              className={`w-6 h-6 cursor-pointer mx-1 ${
                displayCols === cols ? "text-primary" : "text-gray-400"
              }`}
            >
              {cols === "4" ? (
                <path
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  d="M3.75 6A2.25 2.25 0 016 3.75h2.25A2.25 2.25 0 0110.5 6v2.25a2.25 2.25 0 01-2.25 2.25H6a2.25 2.25 0 01-2.25-2.25V6zM3.75 15.75A2.25 2.25 0 016 13.5h2.25a2.25 2.25 0 012.25 2.25V18a2.25 2.25 0 01-2.25 2.25H6A2.25 2.25 0 013.75 18v-2.25zM13.5 6a2.25 2.25 0 012.25-2.25H18A2.25 2.25 0 0120.25 6v2.25A2.25 2.25 0 0118 10.5h-2.25a2.25 2.25 0 01-2.25-2.25V6zM13.5 15.75a2.25 2.25 0 012.25-2.25H18a2.25 2.25 0 012.25 2.25V18A2.25 2.25 0 0118 20.25h-2.25A2.25 2.25 0 0113.5 18v-2.25z"
                />
              ) : cols === "6" ? (
                <path
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  d="M8.25 6.75h12M8.25 12h12m-12 5.25h12M3.75 6.75h.007v.008H3.75V6.75zm.375 0a.375.375 0 11-.75 0 .375.375 0 01.75 0zM3.75 12h.007v.008H3.75V12zm.375 0a.375.375 0 11-.75 0 .375.375 0 01.75 0zm-.375 5.25h.007v.008H3.75v-.008zm.375 0a.375.375 0 11-.75 0 .375.375 0 01.75 0z"
                />
              ) : (
                <path
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  d="M3.375 19.5h17.25m-17.25 0a1.125 1.125 0 01-1.125-1.125M3.375 19.5h7.5c.621 0 1.125-.504 1.125-1.125m-9.75 0V5.625m0 12.75v-1.5c0-.621.504-1.125 1.125-1.125m18.375 2.625V5.625m0 12.75c0 .621-.504 1.125-1.125 1.125m1.125-1.125v-1.5c0-.621-.504-1.125-1.125-1.125m0 3.75h-7.5A1.125 1.125 0 0112 18.375m9.75-12.75c0-.621-.504-1.125-1.125-1.125H3.375c-.621 0-1.125.504-1.125 1.125m19.5 0v1.5c0 .621-.504 1.125-1.125 1.125M2.25 5.625v1.5c0 .621.504 1.125 1.125 1.125m0 0h17.25m-17.25 0h7.5c.621 0 1.125.504 1.125 1.125M3.375 8.25c-.621 0-1.125.504-1.125 1.125v1.5c0 .621.504 1.125 1.125 1.125m17.25-3.75h-7.5c-.621 0-1.125.504-1.125 1.125m8.625-1.125c.621 0 1.125.504 1.125 1.125v1.5c0 .621-.504 1.125-1.125 1.125m-17.25 0h7.5m-7.5 0c-.621 0-1.125.504-1.125 1.125v1.5c0 .621.504 1.125 1.125 1.125M12 10.875v-1.5m0 1.5c0 .621-.504 1.125-1.125 1.125M12 10.875c0 .621.504 1.125 1.125 1.125m-2.25 0c.621 0 1.125.504 1.125 1.125M13.125 12h7.5m-7.5 0c-.621 0-1.125.504-1.125 1.125M20.625 12c.621 0 1.125.504 1.125 1.125v1.5c0 .621-.504 1.125-1.125 1.125m-17.25 0h7.5M12 14.625v-1.5m0 1.5c0 .621-.504 1.125-1.125 1.125M12 14.625c0 .621.504 1.125 1.125 1.125m-2.25 0c.621 0 1.125.504 1.125 1.125m0 1.5v-1.5m0 0c0-.621.504-1.125 1.125-1.125m0 0h7.5"
                />
              )}
            </svg>
          ))}
        </div>
        <div className="mt-4 md:mt-0 flex items-center">
          <p className="text-sm mr-2 text-gray-500">limit:</p>
          <Dropdown
            value={limit}
            onChange={handleLimitChange}
            options={["All", "10", "20", "30", "40"]}
          />
        </div>
      </div>

      <div className="flex items-center mt-4 md:mt-0">
        <svg
          className="w-6 h-6 mr-3 text-gray-500 "
          aria-hidden="true"
          xmlns="http://www.w3.org/2000/svg"
          fill="currentColor"
          viewBox="0 0 18 20"
        >
          <path
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            d="M7.75 4H19M7.75 4a2.25 2.25 0 0 1-4.5 0m4.5 0a2.25 2.25 0 0 0-4.5 0M1 4h2.25m13.5 6H19m-2.25 0a2.25 2.25 0 0 1-4.5 0m4.5 0a2.25 2.25 0 0 0-4.5 0M1 10h11.25m-4.5 6H19M7.75 16a2.25 2.25 0 0 1-4.5 0m4.5 0a2.25 2.25 0 0 0-4.5 0M1 16h2.25"
          />
        </svg>
        <Dropdown
          value={sortType}
          onChange={handleSortTypeChange}
          options={Object.values(SortOptions)}
        />
      </div>
    </div>
  );
};
