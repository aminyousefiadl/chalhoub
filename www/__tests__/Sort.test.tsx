import { Sort } from "@/components";
import { render } from "@testing-library/react";

const setup = () => ({
  ...render(<Sort />),
});

describe("Acceptable Sort Component", () => {
  it("renders to match snapshot", () => {
    const { baseElement } = setup();
    expect(baseElement).toMatchSnapshot();
  });
});
