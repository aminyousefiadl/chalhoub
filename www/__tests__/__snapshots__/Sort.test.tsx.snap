// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Acceptable Sort Component renders to match snapshot 1`] = `
<body>
  <div>
    <div
      class="bg-white md:sticky top-0 rounded-b-lg z-20 mt-16 md:mt-0 md:flex items-center justify-between p-4 w-full shadow-lg mb-8"
    >
      <div
        class="md:flex items-center"
      >
        <div
          class="flex items-center mr-8"
        >
          <svg
            class="w-6 h-6 cursor-pointer mx-1 text-primary"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3.75 6A2.25 2.25 0 016 3.75h2.25A2.25 2.25 0 0110.5 6v2.25a2.25 2.25 0 01-2.25 2.25H6a2.25 2.25 0 01-2.25-2.25V6zM3.75 15.75A2.25 2.25 0 016 13.5h2.25a2.25 2.25 0 012.25 2.25V18a2.25 2.25 0 01-2.25 2.25H6A2.25 2.25 0 013.75 18v-2.25zM13.5 6a2.25 2.25 0 012.25-2.25H18A2.25 2.25 0 0120.25 6v2.25A2.25 2.25 0 0118 10.5h-2.25a2.25 2.25 0 01-2.25-2.25V6zM13.5 15.75a2.25 2.25 0 012.25-2.25H18a2.25 2.25 0 012.25 2.25V18A2.25 2.25 0 0118 20.25h-2.25A2.25 2.25 0 0113.5 18v-2.25z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <svg
            class="w-6 h-6 cursor-pointer mx-1 text-gray-400"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.25 6.75h12M8.25 12h12m-12 5.25h12M3.75 6.75h.007v.008H3.75V6.75zm.375 0a.375.375 0 11-.75 0 .375.375 0 01.75 0zM3.75 12h.007v.008H3.75V12zm.375 0a.375.375 0 11-.75 0 .375.375 0 01.75 0zm-.375 5.25h.007v.008H3.75v-.008zm.375 0a.375.375 0 11-.75 0 .375.375 0 01.75 0z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <svg
            class="w-6 h-6 cursor-pointer mx-1 text-gray-400"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3.375 19.5h17.25m-17.25 0a1.125 1.125 0 01-1.125-1.125M3.375 19.5h7.5c.621 0 1.125-.504 1.125-1.125m-9.75 0V5.625m0 12.75v-1.5c0-.621.504-1.125 1.125-1.125m18.375 2.625V5.625m0 12.75c0 .621-.504 1.125-1.125 1.125m1.125-1.125v-1.5c0-.621-.504-1.125-1.125-1.125m0 3.75h-7.5A1.125 1.125 0 0112 18.375m9.75-12.75c0-.621-.504-1.125-1.125-1.125H3.375c-.621 0-1.125.504-1.125 1.125m19.5 0v1.5c0 .621-.504 1.125-1.125 1.125M2.25 5.625v1.5c0 .621.504 1.125 1.125 1.125m0 0h17.25m-17.25 0h7.5c.621 0 1.125.504 1.125 1.125M3.375 8.25c-.621 0-1.125.504-1.125 1.125v1.5c0 .621.504 1.125 1.125 1.125m17.25-3.75h-7.5c-.621 0-1.125.504-1.125 1.125m8.625-1.125c.621 0 1.125.504 1.125 1.125v1.5c0 .621-.504 1.125-1.125 1.125m-17.25 0h7.5m-7.5 0c-.621 0-1.125.504-1.125 1.125v1.5c0 .621.504 1.125 1.125 1.125M12 10.875v-1.5m0 1.5c0 .621-.504 1.125-1.125 1.125M12 10.875c0 .621.504 1.125 1.125 1.125m-2.25 0c.621 0 1.125.504 1.125 1.125M13.125 12h7.5m-7.5 0c-.621 0-1.125.504-1.125 1.125M20.625 12c.621 0 1.125.504 1.125 1.125v1.5c0 .621-.504 1.125-1.125 1.125m-17.25 0h7.5M12 14.625v-1.5m0 1.5c0 .621-.504 1.125-1.125 1.125M12 14.625c0 .621.504 1.125 1.125 1.125m-2.25 0c.621 0 1.125.504 1.125 1.125m0 1.5v-1.5m0 0c0-.621.504-1.125 1.125-1.125m0 0h7.5"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </div>
        <div
          class="mt-4 md:mt-0 flex items-center"
        >
          <p
            class="text-sm mr-2 text-gray-500"
          >
            limit:
          </p>
          <select
            class="bg-gray-100 min-w-[6rem] border max-w-xl border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full py-2.5 px-4  "
            id="select"
            placeholder="Select"
          >
            <option
              value="All"
            >
              All
            </option>
            <option
              value="10"
            >
              10
            </option>
            <option
              value="20"
            >
              20
            </option>
            <option
              value="30"
            >
              30
            </option>
            <option
              value="40"
            >
              40
            </option>
          </select>
        </div>
      </div>
      <div
        class="flex items-center mt-4 md:mt-0"
      >
        <svg
          aria-hidden="true"
          class="w-6 h-6 mr-3 text-gray-500 "
          fill="currentColor"
          viewBox="0 0 18 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.75 4H19M7.75 4a2.25 2.25 0 0 1-4.5 0m4.5 0a2.25 2.25 0 0 0-4.5 0M1 4h2.25m13.5 6H19m-2.25 0a2.25 2.25 0 0 1-4.5 0m4.5 0a2.25 2.25 0 0 0-4.5 0M1 10h11.25m-4.5 6H19M7.75 16a2.25 2.25 0 0 1-4.5 0m4.5 0a2.25 2.25 0 0 0-4.5 0M1 16h2.25"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
        <select
          class="bg-gray-100 min-w-[6rem] border max-w-xl border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full py-2.5 px-4  "
          id="select"
          placeholder="Select"
        >
          <option
            value="Recommended"
          >
            Recommended
          </option>
          <option
            value="New"
          >
            New
          </option>
          <option
            value="Cheapest"
          >
            Cheapest
          </option>
        </select>
      </div>
    </div>
  </div>
</body>
`;
